images/hook/update.sh

```diff
@@ -0,0 +1,109 @@
+for promrule in gravity-k8s-rules prometheus-k8s-rules prometheus-nethealth-rules
+do
+    rig delete prometheusrules/$promrule --resource-namespace=monitoring --force
+done
+for servicemonitor in alertmanager coredns grafana kube-apiserver kube-state-metrics kubelet node-exporter prometheus prometheus-operator nethealth
+do
+    rig delete servicemonitors/$servicemonitor --resource-namespace=monitoring --force
+done
+for role in alertmanager-main kube-state-metrics monitoring monitoring:autoscaler monitoring:updater nethealth prometheus-k8s prometheus-k8s-config
+do
+    rig delete roles/"$role" --resource-namespace=monitoring --force
+done
+for rolebinding in alertmanager-main kube-state-metrics monitoring monitoring:autoscaler monitoring:updater nethealth prometheus-k8s prometheus-k8s-config
+do
+    rig delete rolebindings/"$rolebinding" --resource-namespace=monitoring --force
+done
+for clusterrole in kube-state-metrics monitoring:autoscaler monitoring:metrics nethealth node-exporter prometheus-adapter prometheus-k8s prometheus-operator
+do
+    rig delete clusterroles/"$clusterrole" --force
+done
+for clusterrolebinding in kube-state-metrics monitoring:autoscaler monitoring:metrics nethealth node-exporter prometheus-adapter prometheus-k8s prometheus-operator
+do
+    rig delete clusterrolebindings/"$clusterrolebinding" --force
+done
+for crd in alertmanagers.monitoring.coreos.com podmonitors.monitoring.coreos.com prometheuses.monitoring.coreos.com \
+  prometheusrules.monitoring.coreos.com servicemonitors.monitoring.coreos.com
+do
+    rig delete crds/"$crd" --force
+done
+
+echo "---> Checking status"
+rig status "$RIG_CHANGESET" --retry-attempts=120 --retry-period=1s --debug
+
+echo "---> Freezing"
+rig freeze
+
+echo "---> Creating monitoring namespace"
+kubectl apply -f /var/lib/gravity/resources/namespace.yaml
+
+# Generate password for Grafana administrator
+password=$(tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 32 | head -n 1 | tr -d '\n ' | /opt/bin/base64)
+
+/opt/bin/helm3 upgrade --install nethealth --namespace monitoring /var/lib/gravity/resources/charts/nethealth
+/opt/bin/helm3 upgrade --install monitoring --namespace monitoring /var/lib/gravity/resources/charts/kube-prometheus-stack -f /var/lib/gravity/resources/custom-values.yaml \
+    --set grafana.adminPassword="${password}" --set alertmanager.alertmanagerSpec.securityContext.runAsUser="$GRAVITY_SERVICE_USER" --set prometheus.prometheusSpec.securityContext.runAsUser="$GRAVITY_SERVICE_USER"
+/opt/bin/helm3 upgrade --install watcher --namespace monitoring /var/lib/gravity/resources/charts/watcher -f /var/lib/gravity/resources/custom-values-watcher.yaml
+
+# check for readiness of prometheus pod
+timeout 5m sh -c "while ! kubectl --namespace=monitoring get pod prometheus-monitoring-kube-prometheus-prometheus-0; do sleep 10; done"
+kubectl --namespace monitoring wait --for=condition=ready pod prometheus-monitoring-kube-prometheus-prometheus-0
```

Comment:
nit: just curios - does the `timeout` above exist as a precondition for `kubectl wait`?

Reply:
This is the check for the existence of the Kubernetes resource. Otherwise `kubectl wait` fails with `resource does not exist` error.